net: openthread: blank `otPlatLog` when logging is disabled

edmont opened the pull request:
Implement an empty `otPlatLog` function when `CONFIG_LOG` is not
enabled. This also fixes the issue with `log_count_args` not being
available when logging is disabled.

Signed-off-by: Eduardo Montoya <eduardo.montoya@nordicsemi.no>

rlubos reviewed (approved):
LGMT

jukkar commented on subsys/net/lib/openthread/platform/logging.c:
```diff
@@ -39,35 +39,14 @@ static inline int log_translate(otLogLevel aLogLevel)
 
-void otPlatLog(otLogLevel aLogLevel, otLogRegion aLogRegion,
-	       const char *aFormat, ...)
+#if defined(CONFIG_LOG)
+void otPlatLog(otLogLevel aLogLevel, otLogRegion aLogRegion, const char *aFormat, ...)
```
Minor detail, but IMHO it would be better to have only one function definition here so
```
void otPlatLog(otLogLevel aLogLevel, otLogRegion aLogRegion, const char *aFormat, ...)
{
#if defined(CONFIG_LOG)
...
#else 
	ARG_UNUSED(aLogLevel);
	ARG_UNUSED(aLogRegion);
	ARG_UNUSED(aFormat);
#endif 
}
```
This way the function will be always defined in one place even if new parameter is added to it. This way we avoid possible errors if anyone changing forgets to update both definitions. I am not saying that this would happen here, this is more a general comment about the whole code base.

nordic-krch replied:
or even
```
if (!IS_ENABLED(CONFIG_LOG)) {
  return;
}

edmont replied:
Thanks for the suggestions, I picked the most code efficient one.

edmont committed:
net: openthread: blank `otPlatLog` when logging is disabled

Implement an empty `otPlatLog` function when `CONFIG_LOG` is not
enabled. This also fixes the issue with `log_count_args` not being
available when logging is disabled.

Signed-off-by: Eduardo Montoya <eduardo.montoya@nordicsemi.no>
